dot/peerset/peerset_test.go

```diff
@@ -25,96 +29,87 @@ func TestPeerSetBanned(t *testing.T) {
 
 	// we ban a node by setting its reputation under the threshold.
 	rpc := newReputationChange(BannedThresholdValue-1, "")
+
 	// we need one for the message to be processed.
+	processor.EXPECT().Process(Message{Status: Drop, setID: 0x0, PeerID: "testPeer1"})
 	handler.ReportPeer(rpc, peer1)
 	time.Sleep(time.Millisecond * 100)
 
-	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
-
 	// check that an incoming connection from that node gets refused.
+	processor.EXPECT().Process(Message{Status: Reject, setID: 0x0, PeerID: "testPeer1"})
 	handler.Incoming(0, peer1)
-	checkMessageStatus(t, <-ps.resultMsgCh, Reject)
 
 	// wait a bit for the node's reputation to go above the threshold.
 	time.Sleep(time.Millisecond * 1200)
 
 	// try again. This time the node should be accepted.
+	processor.EXPECT().Process(Message{Status: Accept, setID: 0x0, PeerID: "testPeer1"})
 	handler.Incoming(0, peer1)
-	require.NoError(t, err)
-	checkMessageStatus(t, <-ps.resultMsgCh, Accept)
 }
 
 func TestAddReservedPeers(t *testing.T) {
 	t.Parallel()
 
-	handler := newTestPeerSet(t, 0, 2, []peer.ID{bootNode}, []peer.ID{}, false)
+	ctrl := gomock.NewController(t)
+	processor := NewMockMessageProcessor(ctrl)
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: bootNode})
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: reservedPeer})
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: reservedPeer2})
+
+	handler := newTestPeerSet(t, 0, 2, []peer.ID{bootNode}, []peer.ID{}, false, processor)
```

Comment:
I think you would want to check for more things here. Check that reserved peers get added in `handler.peerSet.reservedNode` and noSlotNodes get added.

The previous decoupled code did seem a bit tough, but now after looking at some of the tests it does seem like it was more testable. For example, for this tests, we could have just checked for whether certain message is getting sent or not (previously).